modules/ml/src/main/java/org/apache/ignite/ml/tree/randomforest/RandomForestClassifierTrainer.java

```diff
@@ -17,72 +17,112 @@
 
 /**
- * Random forest classifier trainer.
+ * Classifier trainer based on RandomForest algorithm.
  */
-public class RandomForestClassifierTrainer extends RandomForestTrainer {
+public class RandomForestClassifierTrainer extends RandomForestTrainer<ObjectHistogram<BootstrappedVector>, GiniHistogram, RandomForestClassifierTrainer> {
+    /** Label mapping. */
+    private Map<Double, Integer> lblMapping = new HashMap<>();
+
     /**
-     * Constructs new instance of RandomForestClassifierTrainer.
+     * Constructs an instance of RandomForestClassifierTrainer.
      *
-     * @param predictionsAggregator Predictions aggregator.
-     * @param featureVectorSize Feature vector size.
-     * @param maximumFeaturesCntPerMdl Number of features to draw from original features vector to train each model.
-     * @param ensembleSize Ensemble size.
-     * @param samplePartSizePerMdl Size of sample part in percent to train one model.
-     * @param maxDeep Max decision tree deep.
-     * @param minImpurityDecrease Min impurity decrease.
+     * @param meta Meta.
      */
-    public RandomForestClassifierTrainer(PredictionsAggregator predictionsAggregator,
-        int featureVectorSize,
-        int maximumFeaturesCntPerMdl,
-        int ensembleSize,
-        double samplePartSizePerMdl,
-        int maxDeep,
-        double minImpurityDecrease) {
-
-        super(predictionsAggregator, featureVectorSize, maximumFeaturesCntPerMdl,
-            ensembleSize, samplePartSizePerMdl, maxDeep, minImpurityDecrease);
+    public RandomForestClassifierTrainer(List<FeatureMeta> meta) {
+        super(meta);
     }
 
-    /**
-     * Constructs new instance of RandomForestClassifierTrainer.
-     *
-     * @param featureVectorSize Feature vector size.
-     * @param maximumFeaturesCntPerMdl Number of features to draw from original features vector to train each model.
-     * @param ensembleSize Ensemble size.
-     * @param samplePartSizePerMdl Size of sample part in percent to train one model.
-     * @param maxDeep Max decision tree deep.
-     * @param minImpurityDecrease Min impurity decrease.
-     */
-    public RandomForestClassifierTrainer(int featureVectorSize,
-        int maximumFeaturesCntPerMdl,
-        int ensembleSize,
-        double samplePartSizePerMdl,
-        int maxDeep, double minImpurityDecrease) {
-
-        this(new OnMajorityPredictionsAggregator(), featureVectorSize, maximumFeaturesCntPerMdl,
-            ensembleSize, samplePartSizePerMdl, maxDeep, minImpurityDecrease);
+    /** {@inheritDoc} */
+    @Override protected RandomForestClassifierTrainer instance() {
+        return this;
     }
 
     /** {@inheritDoc} */
-    @Override protected DatasetTrainer<DecisionTreeNode, Double> buildDatasetTrainerForModel() {
-        return new DecisionTreeClassificationTrainer(maxDeep, minImpurityDecrease).withUseIndex(useIndex);
+    @Override protected void init(Dataset<EmptyContext, BootstrappedDatasetPartition> dataset) {
```

Comment:
Please, add comment here to clarify the algorithm's idea (it seems like a popular procedure which can be reused somewhere in algorithms)